Plugins-DB/src/main/java/com/github/unafraid/plugins/DBPluginRepository.java

```diff
@@ -162,4 +170,19 @@ public void uninstallPlugin(AbstractDBPlugin plugin) throws PluginException
 			pluginsDao.delete(dbPlugin.getId());
 		}
 	}
+	
+	/**
+	 * Update plugin's auto start state.
+	 * @param plugin the plugin itself
+	 * @param autoStart if {@code true}, the plugin will start on application boot, if {@code false} the plugin won't
+	 */
+	public void updateAutoStart(AbstractDBPlugin plugin, boolean autoStart)
+	{
+		Objects.requireNonNull(plugin);
+		
+		try (PluginsDAO pluginsDao = DatabaseProvider.DBI.open(PluginsDAO.class))
```

Comment:
Don't do that, make a db check if plugin is present there, u won't even know if u updated or not, check if present if not throw exception if does then proceed